### PR TITLE
Feature/unvendor llvm

### PR DESCRIPTION
**Pull request by ddn0**
We actually don't do anything special with libllvm beyond the stock installation, so it seems reasonable to import it like we import other libraries like boost, aws-sdk-cpp etc.

The biggest hang up is that there is no conan package that supports RTTI for llvm at the moment. I have an outstanding PR to add that option: https://github.com/Manu343726/clang-conan-packages/pull/4

But while we are waiting, we have to pull in this dependency from the system package managers (apt, brew).

If we go this way, we need to document llvm-dev as an external dependency.

I also believe that we should be able to get away with `target_link_libraries(... PRIVATE LLVMSupport)` for most of the libXYZ uses, but overall, the existing CMakeList.txt files are not very precise about their dependencies, so I think I will defer that refactor until we to target-based scheme everywhere.

@insertinterestingnamehere: can you put your comments on the PR here? That way you'll have a chance to review it before it gets merged into ISS or vice versa.

### CODE REVIEW

**Comment by ddn0** [comment]
cc @insertinterestingnamehere

**Comment by insertinterestingnamehere** [comment]
Missed this until just now. I'll look at it tomorrow. Thanks for the heads up!

**Comment by insertinterestingnamehere** [comment]
Looking through this now. The reason we can't have llvm as a dependency of libgalois itself is really because shared object dependencies are transitive so that'd be the same as adding libllvm as a hard dependency of our libraries. It has relatively little to do with how our build is set up.

File: CMakeLists.txt [hunk 1]
```diff
@@ -330,6 +330,9 @@ endif()
 find_package(Boost 1.58.0 REQUIRED COMPONENTS serialization iostreams)
 include_directories(${Boost_INCLUDE_DIRS})
 
+find_package(LLVM REQUIRED CONFIG)
```
Comment (insertinterestingnamehere) [inline CMakeLists.txt:333]: We should still allow building the core libraris (and not the apps) without LLVM, so, for example `make libgalois_shmem` should work without LLVM present. We can tackle that later if needed though.

Comment (ddn0) [reply CMakeLists.txt:333]: Yeah, the only use of llvm in libgalois is in experimental code.

If we get a handle on https://github.com/IntelligentSoftwareSystems/Galois/issues/94 we should also remove the libgalois->libllvm dependency too.

**Comment by ddn0** [comment]
Closed in favor of https://github.com/IntelligentSoftwareSystems/Galois/pull/104

**Pull request closed without merging**